### PR TITLE
Add the release strategy document

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Added the release strategy
- Removed the old release process
- Updated the doc on subscribing to a GitHub release
- Moved the doc on the release process with Prow from the `test-infra` repository

**Related issue(s)**
See also #https://github.com/kyma-project/kyma/issues/125

### CODE REVIEW

**Comment by kazydek** [comment]
#156 #https://github.com/kyma-project/kyma/issues/125

File: guidelines/releases/release-strategy.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+A Technical Writer publishes the blog post on the public Kyma website to announce the release. The post includes release notes prepared based on the input from the Product Owners.
+
+You can learn about the new release from additional notifications published on social media, Slack channels, or announcements made during the Core SIG meetings. Alternatively, you can [subscribe](./release-subscription.md) to GitHub notifications and receive updates each time a new release is available.
+
+## Release scope
+For each of the planning periods described in the release schedule, GitHub epics and issues define and document the release scope with regards to functionality, corrections, and even non-functional requirements. The collection of all documented issues within a release represents the expected scope of that release that the whole organization and all teams define and commit to. The corresponding release in ZenHub identifies all issues and epics that fall under the final release scope.
+
+When planning the release scope, all persons involved in the release must take into consideration the GDPR requirements when processing or storing personal data.
```
Comment (tomekpapiernik) [inline guidelines/releases/release-strategy.md]:
```suggestion
When planning the release scope, all persons involved in the release must take the GDPR requirements into consideration when processing or storing personal data.
```

File: guidelines/releases/release-strategy.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+### Down-porting
```
Comment (tomekpapiernik) [inline guidelines/releases/release-strategy.md]: I think we should use the term "downgrade". "Downporting" doesn't really have a definition and is used scarcely. 

image

vs

image

File: guidelines/releases/release-strategy.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+There is no guaranteed bug fix down-porting for the Kyma project. The default strategy is to upgrade to the latest version. All changes must align with the Kyma upgrade strategy.
```
Comment (tomekpapiernik) [inline guidelines/releases/release-strategy.md]: Does it mean that there's is no established strategy of downgrading that is supposed to fix bugs?

File: guidelines/releases/release-strategy.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+### Release execution
+The Release Engineer executes the release at any time of producing the release artifact. This could result in a release candidate or a final release that is not an automated nightly or weekly build. The release execution begins when the process starts and finishes after publishing all artifacts and the related documentation.
+
+During the execution of a release, Kyma developers run all end-to-end test scenarios. If any single scenario fails, the Release Engineer cannot publish the release.
+
+The Release Manager or Kyma developers notify the Kyma community when the release candidate is available. They pass this information to the public on Slack, requesting the community to validate and provide their short-term feedback. The deadline for the feedback is the day before the planned final release. Ideally, the final release is the same as the last release candidate, avoiding any additional changes. If the Kyma developers identify any release-related issues, they asses them and decide together with the Release Manager if they must fix them and produce a new release candidate.
```
Comment (jose-cortina) [inline guidelines/releases/release-strategy.md]: First line:
> Through the SIG Core, the Kyma community will be notified
Instead of "If the Kyma developers identify any release-related issues..."
> If any release-related issue is reported, it will be assessed to determine whether it would constitute a blocker for the release, in which case the Release Manager will be involved in the decision how to proceed. This will also be communicated to the SIG Core.

File: guidelines/releases/release-strategy.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+### Release publishing
+The final release is available in the GitHub releases, including the installation instructions. It also includes the complete changelog that lists all pull requests merged into this release.
+
+A Technical Writer publishes the blog post on the public Kyma website to announce the release. The post includes release notes prepared based on the input from the Product Owners.
```
Comment (jose-cortina) [inline guidelines/releases/release-strategy.md]: Please add:
> Any known issues will also be documented in the release notes.

File: guidelines/releases/release-strategy.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+## Release scope
+For each of the planning periods described in the release schedule, GitHub epics and issues define and document the release scope with regards to functionality, corrections, and even non-functional requirements. The collection of all documented issues within a release represents the expected scope of that release that the whole organization and all teams define and commit to. The corresponding release in ZenHub identifies all issues and epics that fall under the final release scope.
+
+When planning the release scope, all persons involved in the release must take into consideration the GDPR requirements when processing or storing personal data.
+
+### Down-porting
+There is no guaranteed bug fix down-porting for the Kyma project. The default strategy is to upgrade to the latest version. All changes must align with the Kyma upgrade strategy.
+
+The Kyma team does not plan to provide any patch releases before the first 1.0 production release and encourages the community to always upgrade to the latest release.
```
Comment (jose-cortina) [inline guidelines/releases/release-strategy.md]: Replace "The Kyma team" with "The project Kyma"

File: guidelines/releases/release-strategy.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+  - [Down-porting](#down-porting)
+  - [Deprecation and backward-compatibility](#deprecation-and-backward-compatibility)
+
+## Terminology
+
+### Release types
+
+For release versioning, Kyma follows the approach similar to [Semantic Versioning](https://semver.org/). However, while Semantic Versioning focuses on APIs, the release process for Kyma refers to the complete Kyma deliverable. Despite this difference, Kyma follows the same release types:
+
+- **MAJOR** release to introduce breaking changes
```
[the remark this answers is not in the document]
Comment (kazydek) [reply guidelines/releases/release-strategy.md]: sure, why not

File: guidelines/releases/release-strategy.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+- **MINOR** release to introduce a new functionality in a planned schedule
+- **PATCH** release or a **hot.fix** to introduce a backwards-compatible bug fix as an unscheduled release
+
+>**NOTE:** **Major version zero** refers to the rapid product development that is not production-ready yet. That is why, major changes introduced before the 1.0 release, like in versions 0.5 or 0.6, are not represented as increases in the major version number.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply guidelines/releases/release-strategy.md]: Ok, I changed the sentence a bit.

File: guidelines/releases/release-strategy.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+### Function freeze
+
+The function freeze refers to the point in time after which you cannot merge new features to the release branch.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply guidelines/releases/release-strategy.md]:
```
The **function freeze** term refers to the point in time after which you cannot merge new features to the release branch.

```

File: guidelines/releases/release-strategy.md [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+The Core SIG leaders include the external communication regarding the progress and status of each release to the agenda of the Core SIG bi-weekly meetings. The Release Manager, or a person appointed as a stand-in, presents the release status during those meetings.
+
+### Development end
+This is the last day of the period planned for the release development.
+
+### Release decision
+The Release Manager and the Product Leadership decide to publish the release, based on the status and readiness of the developed functionality. The Release Manager communicates this decision before the release execution starts.
+
+### Release execution
+The Release Engineer executes the release at any time of producing the release artifact. This could result in a release candidate or a final release that is not an automated nightly or weekly build. The release execution begins when the process starts and finishes after publishing all artifacts and the related documentation.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply guidelines/releases/release-strategy.md]: True, I will rephrase it.

File: guidelines/releases/release-strategy.md (hunk 3 above)
Comment (kazydek) [reply guidelines/releases/release-strategy.md]: I rephrased that. Hope it is clear now.

File: guidelines/releases/release-strategy.md (hunk 2 above)
Comment (kazydek) [reply guidelines/releases/release-strategy.md]: I rephrased that a bit.

File: guidelines/releases/release-strategy.md [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+- **Release Engineer** prepares all mandatory release artifacts and conducts technical release activities described in the release process.
+
+- **Release Publisher** prepares and publishes the social media content based on the release notes.
+
+
+## Release requirements and acceptance criteria
+
+### Epics and user stories
+
+The functional implementations that are included in the release are documented as GitHub issues. A given team completes issues according to the Definition of Done (DoD) checklist that is applicable to the product and the team who implements them. Other developers must review your issue and a corresponding Product Owner, Capability Owner, or Quality Owner must accept it.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply guidelines/releases/release-strategy.md]: But! changed it a bit so pls take a look :)

File: guidelines/releases/release-strategy.md [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+
+To accept an issue which does not comply with these requirements, you must receive an exceptional approval from the Product Owner and the Release Manager.
+
+### Test coverage
+
+In Kyma, we follow the accepted test strategy according to which you can verify all functionalities through automated testing that you execute either through the CI or the release pipeline.
+
+The Kyma team must provide the highest possible test coverage to minimize any potential risks and ensure the release is functional and stable.
+
+Whenever possible, we must consider implementing tests that go beyond verifying the functionality of the system. This includes integration, performance, and end-to-end tests.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply guidelines/releases/release-strategy.md]: removed that

File: guidelines/releases/release-strategy.md [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+
+## Release schedule
+A planned or scheduled release follows the planning cycles for the Kyma development that typically take four weeks. At the beginning of each planning cycle, the Head of Capabilities communicates the specific time plan for a release. After reaching the end of the development cycle, the Kyma developers create a release candidate which in most cases is identical to the release published on GitHub.
+
+All planned releases happen at the end of a planned development cycle. Only the Product Lead and Release Manager can delay or resign from such a release if there are reasons that justify such a decision.
+
+We do not schedule patches but rather execute them on-demand if there is a need to address high priority issues before the next planned release. In this context, we explicitly define high priority issues as those which either affect multiple Kyma installations or result in the financial loss for the users. The Release Manager is involved in the decision to execute a patch release. However, we do not expect to have patch releases before the official Kyma 1.0 version.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply guidelines/releases/release-strategy.md]: I rephrased that.

File: guidelines/releases/release-strategy.md (hunk 4 above)
Comment (kazydek) [reply guidelines/releases/release-strategy.md]: I will strive to make it as "active" as possible and remove any passive voice from it (we try to avoid it in docs), preserving the meaning ;)

Comment (kazydek) [reply guidelines/releases/release-strategy.md]: ...although I don't get the Core SIg involvement here?

Comment (kazydek) [reply guidelines/releases/release-strategy.md]: I rephrased that a bit - pls verify if that version is fine :)

File: guidelines/releases/release-strategy.md [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+The Core SIG leaders include the external communication regarding the progress and status of each release to the agenda of the Core SIG bi-weekly meetings. The Release Manager, or a person appointed as a stand-in, presents the release status during those meetings.
+
+### Development end
+This is the last day of the period planned for the release development.
+
+### Release decision
+The Release Manager and the Product Leadership decide to publish the release, based on the status and readiness of the developed functionality. The Release Manager communicates this decision before the release execution starts.
+
+### Release execution
+The release execution is a manual process and is not part of an automated nightly or weekly build. It begins when the Release Engineer creates a release branch with all release artifacts. The process finishes after publishing both the artifacts and the related documentation as a release candidate that can be promoted as a final release.
```
Comment (tomekpapiernik) [inline guidelines/releases/release-strategy.md]:
```suggestion
The release execution is a manual process and is not part of an automated nightly or weekly build. It begins when the Release Engineer creates a release branch with all release artifacts. The process finishes after publishing both the artifacts and the related documentation as a release candidate that can be promoted to a final release.
```

File: guidelines/releases/release-strategy.md [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+
+## Release scope
+For each of the planning periods described in the release schedule, GitHub epics and issues define and document the release scope with regards to functionality, corrections, and even non-functional requirements. The collection of all documented issues within a release represents the expected scope of that release that the whole organization and all teams define and commit to. The corresponding release in ZenHub identifies all issues and epics that fall under the final release scope.
+
+When planning the release scope, all persons involved in the release must take the GDPR requirements into consideration when processing or storing personal data.
+
+### Version maintenance
+There is no guaranteed support for the bug fixes in the previous Kyma versions. The default strategy is to upgrade to the latest version.
+
+The Kyma project does not plan to provide any patch releases before the first 1.0 production release and encourages the community to always upgrade to the latest release.
```
Comment (tomekpapiernik) [inline guidelines/releases/release-strategy.md]: `first 1.0 release` sounds like there's gonna be multiple 1.0 releases. I'd skip "first".

File: guidelines/releases/release-strategy.md [hunk 16]
```diff
@@ -0,0 +1,168 @@
...
+The Kyma community must provide the highest possible test coverage to minimize any potential risks and ensure the release is functional and stable.
+
+Whenever possible, consider implementing tests that go beyond verifying the functionality of the system. This includes integration, performance, and end-to-end tests.
+
+### Compliance rules
+
+All Kyma organization members involved in the release process comply with contribution rules defined for Kyma and ensure that all checks built into the continuous integration process pass. These requirements represent the absolute minimal acceptance criteria needed for a release from a compliance perspective. These rules can change as Kyma approaches the 1.0 release.
+
+Additionally, manual pre-release verification must cover these areas:
+- the use of the third-party software and components that require prior approval
```
Comment (jose-cortina) [inline guidelines/releases/release-strategy.md]: "prior approval" refers to an internal process. Maybe rephrase to "licensing"

File: guidelines/releases/release-strategy.md [hunk 17]
```diff
@@ -0,0 +1,168 @@
...
+- the use and storage of personal data subject to the GDPR requirements
+
+### Security
+
+Open-source projects, like any other, must ensure secure development. You can provide it by constantly raising the awareness of everyone involved in the project. In Kyma, we strive to make all developers aware of the secure development requirement. For example, we conduct threat modeling workshops during which the participants proactively think about possible security threats in the architecture, infrastructure, and implementation.
+
+The Release Manager in Kyma takes care of formal security validation activities performed before major releases. The results of these activities influence the release decision. Lack of attention to security topics can result in the release delay.
```
Comment (jose-cortina) [inline guidelines/releases/release-strategy.md]: "the" release delay?
